edb/server/protocol/ai_ext.py

```diff
@@ -363,6 +378,264 @@ async def handle_request(
         return
 
 
+async def _handle_rag_request(
+    protocol: protocol.HttpProtocol,
+    request: protocol.HttpRequest,
+    response: protocol.HttpResponse,
+    db: dbview.Database,
+    tenant: srv_tenant.Tenant,
+) -> None:
+    try:
+        body = json.loads(request.body)
+        if not isinstance(body, dict):
+            raise TypeError(
+                'the body of the request must be a JSON object')
+
+        context = body.get('context')
+        if context is None:
+            raise TypeError(
+                'missing required "context" object in request')
+        if not isinstance(context, dict):
+            raise TypeError(
+                '"context" value in request is not a valid JSON object')
+
+        ctx_query = context.get("query")
+        ctx_variables = context.get("variables")
+        ctx_globals = context.get("globals")
+        ctx_max_obj_count = context.get("max_object_count")
+
+        if not ctx_query:
+            raise TypeError(
+                'missing required "query" in request "context" object')
```

Comment:
Should we make sure that it parses as a standalone fragment?

Reply:
why?  If it parses while wrapped then that's good enough, no?

Reply:
I worry that some sort of injection might be possible.

Reply:
Another thing we need to handle is comments in the query.

For that it probably suffices to append a newline after?

Reply:
The API here isn't intended to be used by untrusted parties, it is similar in that regard to edgeql+http or graphql.  We also explicitly disable all capabilities, including mutation.

Reply:
Maybe my answer is just "it squicks me out", then

Reply:
I think there are queries with unbalanced parens that would work